telethon/client/uploads.py

```diff
@@ -393,18 +393,19 @@ class UploadMethods(ButtonMethods, MessageParseMethods, UserMethods):
         file_handle = None
         as_image = utils.is_image(file) and not force_document
         use_cache = types.InputPhoto if as_image else types.InputDocument
-        if not isinstance(file, str):
-            file_handle = await self.upload_file(
-                file, progress_callback=progress_callback,
-                use_cache=use_cache if allow_cache else None
-            )
-        elif re.match('https?://', file):
+
+        if re.match('https?://', file):
```

Comment:
I'm fairly confident this will raise with non-`str`.

Reply:
You are right. Why did I think it wouldn't raise..? Maybe I was too sleepy last night. I'll fix it right away.